src/routes/auth.js

```diff
@@ -388,3 +390,34 @@ router.post(`/sudoTestCommand/:secret/makeadminforclient`, async (req, res) => {
         });
     }
 });
+
+router.post('/requestToken', async (req, res) => {
+    try {
+        const { q } = req.body;
+        const { clientId } = decode(q);
+        const client = await Client.findById(clientId);
+
+        if (!client) {
+            return res.status(400).json({
+                err: true,
+                msg: 'No client found',
+            });
+        }
+
+        verify(q, client.access_token, {
+            algorithms: ['HS256'],
+        });
+        const requestToken = makeid(64, true);
+        rtoken.hmset(requestToken.toString(), { cId: clientId });
+        rtoken.expire(requestToken.toString(), keys.reqExpTime);
```

Comment:
Could you briefly explain what does `rtoken.expire` do? Does it auto-delete the token from the map once a timeout occurs?

Also have you tested these methods?

Reply:
Yes, this method removes the token from the map after the given period of time. And yes, I have tested all these methods by sending api requests through Postman and creating dummy routes.